contracts/implementation/PoolFactory.sol

```diff
@@ -80,16 +86,32 @@ contract PoolFactory is IPoolFactory, Ownable {
     }
 
     /**
-     * @notice Deploy a leveraged pool with given parameters
+     * @notice Deploy a leveraged pool and its committer/pool tokens with given parameters
      * @param deploymentParameters Deployment parameters of the market. Some may be reconfigurable
      * @return Address of the created pool
      */
-    function deployPool(PoolDeployment calldata deploymentParameters) external override onlyGov returns (address) {
+    function deployPool(PoolDeployment calldata deploymentParameters) external override returns (address) {
         address _poolKeeper = address(poolKeeper);
         require(_poolKeeper != address(0), "PoolKeeper not set");
-        require(address(poolCommitterDeployer) != address(0), "PoolCommitterDeployer not set");
+        require(
+            IOracleWrapper(deploymentParameters.oracleWrapper).owner() == msg.sender,
+            "Deployer must be oracle wrapper owner"
+        );
+
+        bytes32 uniquePoolHash = keccak256(
+            abi.encode(
+                deploymentParameters.leverageAmount,
+                deploymentParameters.quoteToken,
+                deploymentParameters.oracleWrapper
+            )
```

Comment:
What actually makes this unique? I don't see any checks on uniqueness?

Reply:
theres a unit test madhav made that deploys again with those same parameters u get error. 

if you asking whether we should add more parameters to the hash we had update interval in there as well but madhav removed it and said we can add it later if we want.